pandas/core/computation/expressions.py

```diff
@@ -160,12 +161,12 @@ def _where_numexpr(cond, a, b):
 
 def _has_bool_dtype(x):
     try:
-        return x.dtype == bool
-    except AttributeError:
-        try:
+        if isinstance(x, DataFrame):
```

Comment:
can you use
``from pandas.core.dtypes.generic import ABCDataFrame``
instead

Reply:
@jreback ok! i fixed.